src/main/scala/com/ubirch/decoding/Verify.scala

```diff
@@ -42,34 +23,41 @@ class DefaultVerify(verifier: MultiKeyProtocolVerifier) extends Verify with Lazy
 
       val requestId = record.requestIdHeader().orNull
 
-      record.findHeader(HARDWARE_ID_HEADER_KEY).map(UUID.fromString) match {
-        case Some(hardwareId) =>
-
-          val msgPack = record.value()
+      val hardwareId = Try(
+        record.findHeader(HARDWARE_ID_HEADER_KEY)
+          .map(UUID.fromString)
+          .get
+      ).getOrElse{
+        val errorMsg = s"Header with key $HARDWARE_ID_HEADER_KEY is missing. Cannot verify msgPack."
+        logger.error(errorMsg, v("requestId", requestId))
+        throw new SignatureException(errorMsg)
+      }
 
-          //Todo: Should I check the length of the package before splitting it?
-          val signatureIdentifierLength = Verify.differentiateUbirchMsgPackVersion(msgPack)
-          val restOfMessage = msgPack.dropRight(64 + signatureIdentifierLength)
-          val signature = msgPack.takeRight(64)
+      //Todo: Check the length of package
+      val dataToVerifyAndSignature =
+        Try(MsgPackProtocolDecoder.getDecoder.getDataToVerifyAndSignature(record.value()))
+        .getOrElse{
+          val errorMsg = s"error building message parts for hardwareId $hardwareId."
+          logger.error(errorMsg, v("requestId", requestId))
+          throw new SignatureException("Invalid parts for verification")
+        }
+      val dataToVerify = dataToVerifyAndSignature(0)
+      val signature = dataToVerifyAndSignature(1)
 
-          //Todo: Use cached KeyServiceClient
-          verifier
-            .verifyMulti(hardwareId, restOfMessage, 0, restOfMessage.length, signature) match {
-            case Some(key) =>
-              logger.info(s"signature_verified_for=$hardwareId", v("requestId", requestId))
-              record.withExtraHeaders(("algorithm", key.getSignatureAlgorithm))
-            case None =>
-              val errorMsg = s"signature verification failed for msgPack of hardwareId $hardwareId."
-              logger.error(errorMsg, v("requestId", requestId))
-              throw new SignatureException("Invalid signature")
-          }
+      //Todo: Use cached KeyServiceClient
+      verifier
+        .verifyMulti(hardwareId, dataToVerify, 0, dataToVerify.length, signature) match {
 
+        case Some(key) =>
+          logger.info(s"signature_verified_for=$hardwareId", v("requestId", requestId))
+          record.withExtraHeaders(("algorithm", key.getSignatureAlgorithm))
         case None =>
-          val errorMsg = s"Header with key $HARDWARE_ID_HEADER_KEY is missing. Cannot verify msgPack."
+          val errorMsg = s"signature verification failed for msgPack of hardwareId $hardwareId."
```

Comment:
I think the short name for msgpack is msgpack not msgPack

Reply:
Yes, msgpack is the correct short name

Reply:
Will do the corresponding changes

Reply:
It was resolved.